src/k2/module/k23si/Module.h

```diff
@@ -129,6 +129,29 @@ class K23SIPartitionModule {
         return result;
     }
 
+    // validate 2 Txns in PUSH requests are within the retention window for the collection. 
+    // return 0 if request is valid. 1 if incumbent mtr is stale, 2 if incubent mtr is stale, 3 if both are stale.
+    template <typename RequestT>
+    uint8_t _validatePushRetention(const RequestT& req) const {
+        bool result1 = req.incumbentMTR.timestamp.compareCertain(_retentionTimestamp) >= 0;
```

Comment:
We only need to validate that the challengerMTR is within the retention window. If the incumbentMTR is outside the retention window then the desired behavior is that the PUSH will go through and the incumbent will be evicted.
So we can also simplify the return of this _validate function to be a bool like the others.